presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/SelectorSpec.java

```diff
@@ -61,21 +68,35 @@ public boolean equals(Object other)
         if (other == this) {
             return true;
         }
+
         if (!(other instanceof SelectorSpec)) {
             return false;
         }
+
         SelectorSpec that = (SelectorSpec) other;
-        return (group.equals(that.group) &&
+
+        boolean metadataGroupsEqual;
+        if (groups.getMetadata().isPresent()) {
+            metadataGroupsEqual = that.groups.getMetadata().isPresent() && that.groups.getMetadata().get().equals(groups.getMetadata().get());
+        }
+        else {
+            metadataGroupsEqual = !that.groups.getMetadata().isPresent();
+        }
+
+        return (groups.getData().equals(that.groups.getData()) &&
+                metadataGroupsEqual &&
```

Comment:
`groups.getMetadata().equals(that.groups.getMetadata())` should be equivalent to your `if-else`?